### ISSUE TITLE
v0.17.1: `no-unused-vars` does not report unused import specifiers

### ISSUE THREAD

**Issue by benmosher** [eslint/eslint#2114]
For the following test code, `x` will not be reported as unused, but `a` will.

Test code:
```javascript
import x from "y";

var a = "b";

```

Sibling `.eslintrc` to allow `import` syntax:
```yaml
---
ecmaFeatures:
  modules: true
```

Console output:
```
benm-rmbp:pristine ben$ $(npm bin)/eslint test.js

test.js
  3:4  error  a is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

(failing) Test added to `invalid` array of `tests/lib/rules/no-unused-vars.js`:
```javascript
{ code: "import a from 'b';",
  errors: [{ message: "a is defined but never used", type: "Identifier"}], 
  ecmaFeatures: {modules: true} }
```

That test fails for stock setup (as of commit 1626126), but passes with a local `npm install escope@2.0.7`. (which then breaks one other test.)

**Comment by nzakas** [comment]
This is partly related to escope issues, so likely will be fixed when we upgrade.

**Comment by jbrumwell** [comment]
+1

**Comment by wyuenho** [comment]
+1

**Comment by nzakas** [comment]
+1's aren't very useful to us. This issue is open, so it still be addressed. Instead of leaving a +1, why not take a look at this or other issues and chip in to help close them? http://eslint.org/docs/developer-guide/contributing.html

**Comment by OliverJAsh** [comment]
:+1:

**Comment by nzakas** [comment]
Fixed by #2235.
